asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -236,3 +227,42 @@ def make_df_from_docking_results(results: list["DockingResult"]):
         import pandas as pd
 
         return pd.DataFrame([r.get_output() for r in results])
+
+    def write_docking_files(self, output_dir: Union[str, Path]):
+        """
+        Write docking files to output_dir
+
+        Parameters
+        ----------
+        output_dir : Union[str, Path]
+            Output directory
+        """
+        self._write_docking_files(self, output_dir)
+
+    @staticmethod
+    def _write_docking_files(result: "DockingResult", output_dir: Union[str, Path]):
+        output_dir = Path(output_dir)
+        output_pref = result.unique_name()
+        compound_dir = output_dir / output_pref
+        compound_dir.mkdir(parents=True, exist_ok=True)
+        output_sdf_file = compound_dir / "docked.sdf"
+        output_pdb_file = compound_dir / "docked_complex.pdb"
+        output_json_file = compound_dir / "docking_result.json"
+        result.posed_ligand.to_sdf(output_sdf_file)
+        combined_oemol = result.to_posed_oemol()
+        save_openeye_pdb(combined_oemol, output_pdb_file)
+        result.to_json_file(output_json_file)
```

Comment:
Not a blocker. Maybe I'm missing something, but is there a reason why we need to write our own `to_json_file` implementation insetad of using [pydantic's one](https://docs.pydantic.dev/1.10/usage/exporting_models/#modeljson)?

Reply:
Ah just lightly wraps the `json()` call from pydantic in writing to a file AFAIK the linked function just dumps the json in memory.